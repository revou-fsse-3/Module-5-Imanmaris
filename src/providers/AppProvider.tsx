
import { ReactNode, createContext, useState } from "react";

interface Forecast {
    date: string;
    date_epoch:number;
}

interface Props {
    children: ReactNode;
}

interface Context {
    forecast?: Forecast;
    setForecast?: React.Dispatch<React.SetStateAction<Forecast | undefined>>;
}

const defaultValue: Context = {
    forecast: undefined,
    setForecast: undefined
}

export const AppContext = createContext(defaultValue);

const AppProvider = ({children}: Props) => {
    const [forecast, setForecast] = useState<Forecast>();

    return (
        <AppContext.Provider value= {{ forecast, setForecast}}>
            {children}
        </AppContext.Provider>
    )
}

export default AppProvider