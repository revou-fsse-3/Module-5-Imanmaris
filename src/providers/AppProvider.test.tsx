
import {screen, fireEvent, render, waitFor} from "@testing-library/react";
import AppProvider, { useUser } from "./AppProvider";

describe('AppProvider unit testing', () => {
    
    test('AppProvider should be return correct value', async () => {
        const dummyUser = {
            date: '27-01-2024',
            date_epoch: 27012024
        }

        const TestComponent = () => {
        
            const context = useUser()
            const { forecast, setForecast } = context

            const getUserData = () => {
                setForecast?.(dummyUser)
            }

            return (
                <>
                    <div>
                        <p>{`Name: ${forecast?.date}`}</p>
                        <button onClick={getUserData}>New User</button>
                    </div>  
                </>
            )
        };

        render(
            <AppProvider>
                <TestComponent/>
            </AppProvider>
        )

      expect(screen.getByText('New User')).toBeDefined();
      await waitFor (() => fireEvent.click(screen.getByRole('button')))
      expect(screen.getByText(`Name: ${dummyUser.date}`)).toBeDefined();

    });
  
});