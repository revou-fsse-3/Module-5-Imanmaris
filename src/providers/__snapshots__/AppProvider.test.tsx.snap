// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppProvider unit testing AppProvider should be return correct value 1`] = `
<button>
  New Location
</button>
`;

exports[`AppProvider unit testing AppProvider should be return correct value 2`] = `
<p>
  Date: 27-01-2024
</p>
`;

exports[`AppProvider unit testing AppProvider should be return correct value 3`] = `
<p>
  Date Epoch: 27012024
</p>
`;
